runconfig/config.go

```diff
@@ -134,28 +134,29 @@ func NewCommand(parts ...string) *Command {
-	Cmd             *Command              // Command to run when starting the container
-	Image           string                // Name of the image as it was passed by the operator (eg. could be symbolic)
-	Volumes         map[string]struct{}   // List of volumes (mounts) used for the container
-	VolumeDriver    string                // Name of the volume driver used to mount volumes
-	WorkingDir      string                // Current directory (PWD) in the command will be launched
-	Entrypoint      *Entrypoint           // Entrypoint to run when starting the container
-	NetworkDisabled bool                  // Is network disabled
-	MacAddress      string                // Mac Address of the container
-	OnBuild         []string              // ONBUILD metadata that were defined on the image Dockerfile
-	Labels          map[string]string     // List of labels set to this container
+	Hostname         string                // Hostname
+	Domainname       string                // Domainname
+	User             string                // User that will run the command(s) inside the container
+	AttachStdin      bool                  // Attach the standard input, makes possible user interaction
+	AttachStdout     bool                  // Attach the standard output
+	AttachStderr     bool                  // Attach the standard error
+	ExposedPorts     map[nat.Port]struct{} // List of exposed ports
+	PublishService   string                // Name of the network service exposed by the container
+	Tty              bool                  // Attach standard streams to a tty, including stdin if it is not closed.
+	OpenStdin        bool                  // Open stdin
+	StdinOnce        bool                  // If true, close stdin after the 1 attached client disconnects.
+	Env              []string              // List of environment variable to set in the container
+	Cmd              *Command              // Command to run when starting the container
+	Image            string                // Name of the image as it was passed by the operator (eg. could be symbolic)
+	Volumes          map[string]struct{}   // List of volumes (mounts) used for the container
+	VolumeDriver     string                // Name of the volume driver used to mount volumes
+	WorkingDir       string                // Current directory (PWD) in the command will be launched
+	Entrypoint       *Entrypoint           // Entrypoint to run when starting the container
+	NetworkDisabled  bool                  // Is network disabled
+	MacAddress       string                // Mac Address of the container
+	OnBuild          []string              // ONBUILD metadata that were defined on the image Dockerfile
+	Labels           map[string]string     // List of labels set to this container
+	TrustedBuildArgs map[string][]string   // list of build-time args that are allowed for expansion/substitution and passing to commands in 'run'.
```

Comment:
ok, makes sense.

I used `TrustedBuildArgs` just to bring out the fact that this is the list of variables (defined using `ARG`) which a Dockerfile author trusts it's ok to pass values to the builder.

We can also call them `AllowedBuildArgs` or `DefinedBuildArgs`? WDYT